### ISSUE TITLE
aShuffle playlist on save

### ISSUE THREAD

**Issue by Rouzax** [WritheM/IntelliMerge#22]
**Is your feature request related to a problem? Please describe.**
Somehow even when I have playlists with over a 1000 songs I will often here the same repeated songs from that playlist. Think this has to do with the broken shuffle of Spotify.

**Describe the solution you'd like**
Could IntelliMerge shuffle the playlist on safe?

**Comment by pironic** [comment]
When it comes to the 'random-ness' of spotify's shuffle I completely agree that it is severely lacking. I'm not sure if there is an algorithm that would explain why it isn't random.. I've played my 3200 track playlist and heard the same song twice while on random. As for Intellimerge being able to solve their deficiency, I love the idea... however...

I fear this might be similar to #21 in that it is very difficult to sort/shuffle a playlist.

BUT! We could limit the frequency that this happens by calling it manually and restricting to a VIP only feature. Additionally, it might be worth re-opening #21 with the same limitations. If we give VIP's the option to click a single button on the site to sort/shuffle but not schedule it, that could be a solution to the numerous API hits to the Spotify server.

I will think about this a little more and research how to pull this off... Thank you for the suggestion @Rouzax! Keep 'em coming!

**Comment by Rouzax** [comment]
Could you then maybe do it automatically once every week or so?

It does help to disable this setting in Spotify to have more randomness.
![Screenshot_20200912-090646_Spotify](https://user-images.githubusercontent.com/4103090/92989957-5ab2c500-f4d8-11ea-888c-6c63f59de484.jpg)

**Comment by pironic** [comment]
no guarantees... but maybe?

thank you, i'll try that setting.

**Comment by Rouzax** [comment]
Wondering if you have this in the pipeline.

**Comment by pironic** [comment]
At the moment I do not. I'm currently engaged in a different full-time project that will take my time until ~Jan '22. I still do maintenance on Intellimerge right now, but no feature additions at the moment. I'm sorry :(

**Comment by Rouzax** [comment]
Understand you haven't got the time yet just wanted to make you aware that a lot of people are really fed up with Spotify shuffle.
https://community.spotify.com/t5/Live-Ideas/All-Platforms-Option-to-have-a-true-shuffle/idi-p/4880594
This might be a great way to expand the premium subscription.

**Comment by Rouzax** [comment]
Since there is active development again, any change for this?

**Comment by pironic** [comment]
I've done some research into all the ways i can accommodate this... let me ask your opinion @Rouzax: if it meant that the "Date Added" column would always show the date that intellimerge last ran would you still want it? The only options are 1 api hit per track to tell spotify the new location for a track, or to rebuild the entire track in batches of 100 at a time which always rebuilds the playlist. 

Either way it's not great... if you had a playlist with 5000 songs in it, i still have to hit the spotify api 50 times to rebuild the playlist every hour. That chews through a lot of our rate limits as set by spotify. even if we limit it to the vip users, that's still a lot. I'm not sure Spotify would be thrilled of all those extra requests. :/

**Comment by Rouzax** [comment]
The date added would not matter for me.
I would also not want/need to have shuffle on the same rate as adding new tracks, once a day for shuffle for instance would already be more than enough.
At the moment I run https://stevenaleong.com/tools/spotifyplaylistrandomizer every month or something

**Comment by pironic** [comment]
yeah. i like the dates too... if those gotta stay, then it's gonna be a single api PER SONG. which is just not realistic at any interval... even entirely manual. 5000 hits for a single playlist? yikes. i'm not sure how steven is doing it... unless he's just doing multiple hits and saying to hell with the rate limits. which seems plausible based on the text on his site:

> If it throws an error, it could possibly be caused by too many requests being sent to Spotify.


I did notice that spotify has implemented an alegedly more random option that includes "less repeats" though. maybe we need to rely on steven or that setting, as we have.

**Comment by Rouzax** [comment]
> yeah. i like the dates too... if those gotta stay, then it's gonna be a single api PER SONG. which is just not realistic at any interval... even entirely manual. 5000 hits for a single playlist? yikes. i'm not sure how steven is doing it... unless he's just doing multiple hits and saying to hell with the rate limits. which seems plausible based on the text on his site:
> 
> > If it throws an error, it could possibly be caused by too many requests being sent to Spotify.
> 
> I did notice that spotify has implemented an alegedly more random option that includes "less repeats" though. maybe we need to rely on steven or that setting, as we have.

Sorry, I meant. I don’t care about the added to dates.
I forgot to add the not 😄
